lib/Models/UrthecastCatalogGroup.js

```diff
@@ -0,0 +1,140 @@
+'use strict';
+
+/*global require*/
+var defineProperties = require('terriajs-cesium/Source/Core/defineProperties');
+var knockout = require('terriajs-cesium/Source/ThirdParty/knockout');
+
+var ModelError = require('./ModelError');
+var raiseErrorToUser = require('../Models/raiseErrorToUser');
+var CatalogGroup = require('./CatalogGroup');
+var inherit = require('../Core/inherit');
+var UrthecastCatalogItem = require('./UrthecastCatalogItem');
+var urthecastClient = require('urthecast');
+
+/**
+ * A {@link CatalogGroup} representing a collection of avalible Urthecast sensor platforms.
+ *
+ * @alias UrthecastCatalogGroup
+ * @constructor
+ * @extends CatalogGroup
+ *
+ * @param {Terria} terria The Terria instance.
+ */
+var UrthecastCatalogGroup = function(terria) {
+    CatalogGroup.call(this, terria);
+
+    /**
+     * Gets or sets a hash of names of blacklisted data layers.  A layer that appears in this hash
+     * will not be shown to the user.  In this hash, the keys should be the Title of the layers to blacklist,
+     * and the values should be "true".  This property is observable.
+     * @type {Object}
+     */
+    this.blacklist = undefined;
+
+    knockout.track(this, ['blacklist']);
+};
+
+inherit(CatalogGroup, UrthecastCatalogGroup);
+
+defineProperties(UrthecastCatalogGroup.prototype, {
+    /**
+     * Gets the type of data member represented by this instance.
+     * @memberOf UrthecastCatalogGroup.prototype
+     * @type {String}
+     */
+    type : {
+        get : function() {
+            return 'urthecast-group';
+        }
+    },
+
+    /**
+     * Gets a human-readable name for this type of data source, such as 'Urthcast Sensor Platforms'.
+     * @memberOf UrthecastCatalogGroup.prototype
+     * @type {String}
+     */
+    typeName : {
+        get : function() {
+            return 'Urthcast Sensor Platforms';
+        }
+    },
+});
+
+UrthecastCatalogGroup.prototype._load = function() {
+    var catalogGroup = this;
+    var apiKey = catalogGroup.terria.configParameters.urthecastApiKey;
```

Comment:
We should inform the user when the API key and secret are not set, by throwing a `ModelError` with an appropriate `title` and `message`.  It took me a minute to realize why nothing was working.